client/nginx_test.go

```diff
@@ -623,6 +623,62 @@ func TestClientWithHTTPClient(t *testing.T) {
 	}
 }
 
+func TestClientWithMaxAPI(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		apiVersions string
+		expected    int
+	}{
+		{
+			apiVersions: `[4, 5, 6, 7, 8, 9, 25]`,
+			expected:    APIVersion,
+		},
+		{
+			apiVersions: ``,
+			expected:    APIVersion,
+		},
+		{
+			apiVersions: `[4, 5, 6, 7]`,
+			expected:    7,
+		},
+		{
+			apiVersions: `[""]`,
+			expected:    APIVersion,
+		},
+	}
+
+	for _, test := range tests {
+		// Test creating a new client with max API version
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
can you use `t.Run`?

Reply:
Thank you for the review, I've made changes if you can look again. Thank you!